modules/devices/mediaDeviceHelper.js

```diff
@@ -25,6 +25,17 @@ function getNewAudioOutputDevice(newDevices) {
             d.deviceId === selectedAudioOutputDeviceId)) {
         return 'default';
     }
+
+    const settings = APP.store.getState()['features/base/settings'];
```

Comment:
Shouldn't we include this logic to the logic above somehow?

Reply:
Nope, so the logic above is, if the current output device is not 'default' and is not found in the available devices, return 'default', this is in case of removal.

But the newly added logic is if the above is not true (most probably that device is available in the list of devices, so current ouput was not removed), so we check whether our preferred device is available and if yes return it, otherwise keep the logic as before and return undefined.

So you can divide the logic in two: 
- removal, which is the first one 
- addition, the second one.

Reply:
What I mean is, if you remove the current output device shouldn't we try to see if the preferred one is available and choose the preferred one instead of the default? Or this shouldn't be possible?

Reply:
That doesn't make sense. If you have a preferred available and you had removed one you should had removed the preferred, otherwise there is a bug in another place.